Tests/SwiftProtobufTests/Test_JSON.swift

```diff
@@ -481,6 +482,20 @@ class Test_JSON: XCTestCase, PBTestHelpers {
         }
     }
 
+    func testSingleDouble_roundtrip() throws {
+        for _ in 0..<10000 {
+            let d = drand48()
+            assertRoundTripJSON {$0.singleDouble = d}
```

Comment:
If it's wrong, it will only be wrong by `.nextUp` or `.nextDown`, so finding the failing value won't be that hard.

In practice, problems in the formatting code will cause lots of failures; it won't be hard to find problems once we know they're there.